### PR TITLE
Asrc 289

### PR DESCRIPTION
**Pull request by gregoryhull**
289 for review

### CODE REVIEW

File: srcalc/src/main/java/gov/va/med/srcalc/db/RiskModelDao.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package gov.va.med.srcalc.db;
 
+import gov.va.med.srcalc.domain.model.AbstractVariable;
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/db/RiskModelDao.java]: Is this import used?

File: srcalc/src/main/java/gov/va/med/srcalc/domain/model/RiskModel.java [hunk 2]
```diff
@@ -32,6 +33,24 @@
     private Set<ProcedureTerm> fProcedureTerms = new HashSet<>();
     private Set<DerivedTerm> fDerivedTerms = new HashSet<>();
     
+	// Use the same Char set and Reg Expr as for the Variable names.
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/model/RiskModel.java]: Why not just reuse the constants then? See `MultiSelectOption` for an example.

File: srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java [hunk 3]
```diff
@@ -104,7 +104,47 @@ public void saveVariable(final AbstractVariable variable)
     @Transactional(readOnly = true)
     public ImmutableCollection<RiskModel> getAllRiskModels()
     {
-        fLogger.debug("Getting all RiskModels.");
+//        fLogger.debug("Getting all RiskModels.");
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java]: Why is this commented out?

File: srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java [hunk 4]
```diff
@@ -104,7 +104,47 @@ public void saveVariable(final AbstractVariable variable)
...
         return fRiskModelDao.getAllRiskModels();
     }
+    
+    @Override
+    @Transactional(readOnly = true)
+    public RiskModel getRiskModelForId(final int modelId)
+    {
+//        fLogger.debug("Getting RiskModel {}.", modelId);
+        
+        final ImmutableCollection<RiskModel> models = getAllRiskModels();
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java]: This actually does load quite a bit of information from the database. All risk models -> all terms -> all variables, etc. I think it's worth adding a DAO method to just load one risk model.

File: srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java [hunk 5]
```diff
@@ -104,7 +104,47 @@ public void saveVariable(final AbstractVariable variable)
...
+     * Saves the name of the {@link RiskModel} 
+     * @return RiskModel
+     */
+    @Override
+    @Transactional
+    public void saveRiskModel( final RiskModel model ) 
+    {    	
+        fLogger.debug("Saving {}.", model);
+
+        if( getRiskModelForId( model.getId() ) == null )
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java]: What is the point of this check?

File: srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java [hunk 6]
```diff
@@ -104,7 +104,47 @@ public void saveVariable(final AbstractVariable variable)
...
+        	}
+        }
+        
+        fLogger.warn("Unable to find RiskModel for ID {}.", modelId);
+
+        return null;
+    }
+
+    /**
+     * Saves the name of the {@link RiskModel}
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java]: This Javadoc is inaccurate: this method actually saves the whole risk model, including its terms.

File: srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java [hunk 7]
```diff
@@ -104,7 +104,47 @@ public void saveVariable(final AbstractVariable variable)
...
-        fLogger.debug("Getting all RiskModels.");
+//        fLogger.debug("Getting all RiskModels.");
         return fRiskModelDao.getAllRiskModels();
     }
+    
+    @Override
+    @Transactional(readOnly = true)
+    public RiskModel getRiskModelForId(final int modelId)
+    {
+//        fLogger.debug("Getting RiskModel {}.", modelId);
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java]: Why is this commented out?

File: srcalc/src/main/java/gov/va/med/srcalc/service/ModelInspectionService.java [hunk 8]
```diff
@@ -24,4 +24,18 @@
...
+
+    
+    /**
+     * Returns the {@link RiskModel} with the given ID.
+     * @return RiskModelS
+     */
+    public RiskModel getRiskModelForId(int modelId);    
+    
+    /**
+     * Saves the name of the {@link RiskModel}
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/ModelInspectionService.java]: This doc comment is inaccurate: this will actually save the whole RiskModel object, including all of its terms.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/AdminHomeController.java [hunk 9]
```diff
@@ -37,6 +50,19 @@ public AdminHomeController(final AdminService adminService)
     @RequestMapping(method = RequestMethod.GET)
     public String defaultPage(final Model model)
     {
+        final ImmutableCollection<RiskModel>  riskModels = fAdminService.getAllRiskModels();
+
+        // Use a set since there are multiple entries in the risk models list for each model.
+        //
+        Set<EditRiskModel> editRMs = new TreeSet<EditRiskModel>(  );
+
+//        fLogger.debug( "There are {} Risk Models.", riskModels.size());
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/AdminHomeController.java]: I don't understand why you keep commenting out logging statements. One of the advantages of using a logging library is that disabled logging statements (e.g., the statement level is DEBUG but the configured level is INFO) have insignificant performance impact.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/AdminHomeController.java [hunk 10]
```diff
@@ -37,6 +50,19 @@ public AdminHomeController(final AdminService adminService)
...
+
+        for( final RiskModel rm : riskModels ) 
+        {
+        	editRMs.add( EditRiskModel.fromRiskModel( rm ) );
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/AdminHomeController.java]: Why do we need to wrap the RiskModel objects in EditRiskModel objects here? The Risk Models are read-only at this point. We can just use the RiskModel objects themselves as the models for the view.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+@Controller
+@RequestMapping("/admin/models/{riskModelId}")
+public class EditRiskModelController {
+
+    private static final String ATTRIBUTE_RISK_MODEL = "riskModel";
+    
+    private static final Logger fLogger = LoggerFactory.getLogger(EditVariableController.class);
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java]: Incorrect logger name.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    @Inject
+    public EditRiskModelController(final AdminService adminService )
+    {
+        fAdminService = adminService;
+    }
+    
+    @ModelAttribute(ATTRIBUTE_RISK_MODEL)
+    public EditRiskModel createEditRiskModel(
+            @PathVariable final String riskModelId )
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java]: If you change this parameter to an int, Spring will do the type conversion for you.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+            throws InvalidIdentifierException
+    {    	
+    	try {
+        	int modId = Integer.parseInt( riskModelId );
+        	RiskModel rm = fAdminService.getRiskModelForId( modId ); 
+
+        	if( rm == null ) {
+        		throw new InvalidIdentifierException( "Unable to find RiskModel with ID "+riskModelId );
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java]: Good.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+    	try {
+        	int modId = Integer.parseInt( riskModelId );
+        	RiskModel rm = fAdminService.getRiskModelForId( modId ); 
+
+        	if( rm == null ) {
+        		throw new InvalidIdentifierException( "Unable to find RiskModel with ID "+riskModelId );        		
+        	}
+        	EditRiskModel editModel = EditRiskModel.fromRiskModel( rm );
+	
+	    	fLogger.debug("In EditRiskController.createEditRiskModel : found model {} for id.", editModel.getModelName() );
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java]: There is no need to explicitly specify the class and method names here. The logger can be configured to add that automatically if we so desire.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+	        return editModel;    		
+    	}
+    	catch ( NumberFormatException nfe ) {
+    		throw new InvalidIdentifierException( "", nfe );
+    	}
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ModelAndView displayForm(
+            @ModelAttribute(ATTRIBUTE_RISK_MODEL) final EditRiskModel editModel )
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java]: This parameter is unused.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+                    throws InvalidIdentifierException
+    {
+        // Spring has already bound the user input to saveModel; now validate
+        // the input using the appropriate validator.
+        //
+    	saveModel.getValidator().validate( saveModel, bindingResult);
+
+        if( bindingResult.hasErrors() )
+        {
+            fLogger.error( "EditRiskModel has errors: {}", bindingResult);
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java]: "ERROR" level is inappropriate here. See the logging levels in [the Dev Guide](https://github.com/VHAINNOVATIONS/ASRCM/blob/master/docs/devguide.md#enhancing-the-application).

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+import gov.va.med.srcalc.domain.model.Specialty;
+
+public class EditRiskModel implements Comparable<EditRiskModel>{
+
+	private final RiskModel riskModel;
+    private static final Logger fLogger = LoggerFactory.getLogger(EditRiskModel.class);
+
+    private String modelName;
+    // can't change the id so just reference the id from the RiskModel
+    //private final int id;
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: What is the purpose of these commented variables?

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+//    private RiskModel importedModel;
+    
+	protected EditRiskModel( final RiskModel rm ) 
+	{
+		riskModel = rm;
+		modelName = rm.getDisplayName();
+		//id = rm.getId();
+	}
+	
+	public static EditRiskModel fromRiskModel( final RiskModel rm )
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: Nice use of a factory method here.

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.validation.Validator;
+
+import gov.va.med.srcalc.domain.model.RiskModel;
+import gov.va.med.srcalc.domain.model.Specialty;
+
+public class EditRiskModel implements Comparable<EditRiskModel>{
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: This class is missing both class and method Javadocs.

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+	{
+		return modelName;
+	}
+
+	public void setModelName( String mn ) 
+	{
+		modelName = mn;
+	}
+
+	public String getId( )
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: Why do you need to return a string?

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	public String getId( ) 
+	{
+		return Integer.toString( riskModel.getId() );
+	}
+	
+	public List<Specialty> getSpecialties() 
+	{
+		return new ArrayList<Specialty>( );
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: This is OK, but a general tip: `java.util.Collections` provides an optimized `emptyList()` method for this purpose.

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.validation.Validator;
+
+import gov.va.med.srcalc.domain.model.RiskModel;
+import gov.va.med.srcalc.domain.model.Specialty;
+
+public class EditRiskModel implements Comparable<EditRiskModel>{
+
+	private final RiskModel riskModel;
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: By convention, all of our objects' fields are prefixed with `f`. So this would be `fRiskModel`.

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 23]
```diff
@@ -0,0 +1,104 @@
...
+    /**
+     * <p>Returns an appropriate Validator instance for validating this object.
+     * </p>
+     * 
+     * <p>Implementations do not perform their own validation so that the web
+     * code can directly bind user input to these beans without fear of
+     * Exceptions. We instead provide a Spring Validator via this method to
+     * safely ensure that user input meets  class's requirements.</p>
+     */
+    public /*abstract*/ Validator getValidator()
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: As you and I discussed, this method is not necessary. It was only needed for the EditVar classes to support polymorphism. The controller can just directly reference the `EditRiskModelValidator` instead.

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 24]
```diff
@@ -0,0 +1,104 @@
...
+    {
+    	return new EditRiskModelValidator();
+    }
+
+	public String toString() 
+	{
+		return riskModel.toString();
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: Having RiskModel and EditRiskModel both return exactly the same String is confusing: it would be hard to tell which is which in a log. I suggest something like, "EditRiskModel for [RiskModel.toString()]"

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java [hunk 25]
```diff
@@ -0,0 +1,104 @@
...
+		
+//		riskModel = importedModel;
+		
+		return riskModel;
+	}
+	
+    @Override
+    public int compareTo(final EditRiskModel other )
+    {
+    	// Order according to id instead of display name ???
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModel.java]: Indeed, why order by ID?

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidator.java [hunk 26]
```diff
@@ -0,0 +1,50 @@
...
+     * @throws ClassCastException if the given object is not.
+     */
+    @Override
+    public void validate(final Object obj, final Errors e)
+    {
+        final EditRiskModel editRMs = (EditRiskModel)obj;
+        
+//        fLogger.debug("Validating name of EditRiskModel {}", editRMs.getModelName() );
+
+        // Note: the editMDs object is accessible from the e Error object.
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidator.java]: Obsolete variable na e in the comment.

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidator.java [hunk 27]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public boolean supports(final Class<?> clazz)
+    {
+        return EditRiskModel.class.isAssignableFrom(clazz);
+    }
+
+    /**
+     * Validates the given object, using error codes from {@link ValidationCodes}.
+     * @param obj the object to validate. Must be an instance of {@link EditRiskModel}.
+     * @throws ClassCastException if the given object is not.
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidator.java]: If the given object is not... what?

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidator.java [hunk 28]
```diff
@@ -0,0 +1,50 @@
...
+        // Note: the editMDs object is accessible from the e Error object.
+        //
+        ValidationUtils.rejectIfEmpty(e, "modelName", ValidationCodes.NO_VALUE);
+        ValidationUtils2.rejectIfTooLong(e, "modelName", RiskModel.DISPLAY_NAME_MAX );
+        
+        ValidationUtils2.rejectIfDoesntMatch(
+                e, "modelName", RiskModel.VALID_MODEL_NAME_PATTERN,
+                new Object[] {RiskModel.VALID_MODEL_NAME_CHARACTERS});        
+
+        fLogger.debug( " EditRiskModel validates" );
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidator.java]: What is this log line supposed to signify? That the object is valid?

File: srcalc/src/main/webapp/WEB-INF/views/adminHome.jsp [hunk 29]
```diff
@@ -4,6 +4,27 @@
 <srcalc:adminPage title="Model Administration">
 <section>
     <h2>Model Administration</h2>
+    <h3>Model Definitions</h3>
```
Comment (dgtombs) [inline srcalc/src/main/webapp/WEB-INF/views/adminHome.jsp]: Good job on the view.

File: srcalc/src/main/webapp/WEB-INF/views/editRiskModel.jsp [hunk 30]
```diff
@@ -0,0 +1,5 @@
+<%@ taglib uri="/WEB-INF/srcalc.tld" prefix="srcalc" %>
+<srcalc:editModelPage riskModel="${riskModel}"  saveUrl="/admin/models/${riskModel.id}">
```
Comment (dgtombs) [inline srcalc/src/main/webapp/WEB-INF/views/editRiskModel.jsp]: Why did you create a tag for this?

File: srcalc/src/main/java/gov/va/med/srcalc/service/ModelInspectionService.java [hunk 31]
```diff
@@ -24,4 +24,18 @@
...
+     * Returns the {@link RiskModel} with the given ID.
+     * @return RiskModelS
+     */
+    public RiskModel getRiskModelForId(int modelId);    
+    
+    /**
+     * Saves the name of the {@link RiskModel} 
+     * @return RiskModel
+     */
+    public void saveRiskModel( final RiskModel model );
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/ModelInspectionService.java]: This method does not belong in `ModelInspectionService`.

File: srcalc/src/main/java/gov/va/med/srcalc/db/RiskModelDao.java [hunk 32]
```diff
@@ -33,12 +38,30 @@ protected Session getCurrentSession()
      * Returns all RiskModels in the database, in arbitrary order.
      * @return an ImmutableCollection
      */
-    public ImmutableCollection<RiskModel> getAllRiskModels()
+    @SuppressWarnings("unchecked")
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/db/RiskModelDao.java]: This annotation can go specifically on line 45 instead of on the whole method. It is best to keep warnings suppression as isolated as possible.

File: srcalc/src/main/java/gov/va/med/srcalc/db/RiskModelDao.java [hunk 33]
```diff
@@ -33,12 +38,30 @@ protected Session getCurrentSession()
...
         return ImmutableList.copyOf(list);
     }
     
+    /**
+     * Returns the RiskModel in the database for the given Id
+     * @return an RiskModel
+     */
+    public RiskModel getRiskModelForId( final int mid ) 
+    {
+        final Query q =  getCurrentSession().createQuery(
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/db/RiskModelDao.java]: Remember I said Hibernate provides a special-purpose method for this? This whole method body can be replaced with:

    return (RiskModel)getCurrentSession().get(RiskModel.class, mid);

File: srcalc/src/main/java/gov/va/med/srcalc/domain/model/DiscreteTerm.java [hunk 34]
```diff
@@ -135,6 +135,6 @@ public int hashCode()
     public String toString()
     {
         return String.format(
-                "%s=%s?->%s", getVariable(), getOption(), getCoefficient());
+                "%s=%s->%s", getVariable(), getOption(), getCoefficient());
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/model/DiscreteTerm.java]: The question mark was intentional here: it was supposed to imply a boolean condition.

File: srcalc/src/main/java/gov/va/med/srcalc/domain/model/RiskModel.java [hunk 35]
```diff
@@ -17,6 +18,7 @@
  * A risk model definition.
  */
 @Entity
+@Table(name = "risk_model")
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/model/RiskModel.java]: This attribute is unnecessary and having it is inconsistent with other entities. Please drop it.

File: srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java [hunk 36]
```diff
@@ -107,4 +107,30 @@ public void saveVariable(final AbstractVariable variable)
...
+    }
+
+    /**
+     * Saves the {@link RiskModel} 
+     */
+    @Override
+    @Transactional
+    public void saveRiskModel( final RiskModel model ) 
+    {    	
+        if( getRiskModelForId( model.getId() ) == null )
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java]: What is the purpose of this warning?

File: srcalc/src/main/java/gov/va/med/srcalc/service/ModelInspectionService.java [hunk 37]
```diff
@@ -24,4 +24,16 @@
...
+    /**
+     * Returns the {@link RiskModel} with the given ID.
+     * @return RiskModelS
+     */
+    public RiskModel getRiskModelForId(int modelId);    
+    
+    /**
+     * Saves the {@link RiskModel} 
+     */
+    public void saveRiskModel( final RiskModel model );
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/ModelInspectionService.java]: This method still doesn't belong in `ModelInspectionService`. See the interface doc comment: "All returned objects are intended for inspection only. This interface does not provide a means to persist any changes."

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/AdminHomeController.java [hunk 38]
```diff
@@ -37,6 +47,19 @@ public AdminHomeController(final AdminService adminService)
     @RequestMapping(method = RequestMethod.GET)
     public String defaultPage(final Model model)
     {
+        final ImmutableCollection<RiskModel>  riskModels = fAdminService.getAllRiskModels();
+
+        // Use a set since there are multiple entries in the risk models list for each model.
+        //
+        Set<EditRiskModel> editRMs = new TreeSet<EditRiskModel>(  );
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/AdminHomeController.java]: My previous comment still stands: I see no reason to translate from RiskModels to EditRiskModels here. We can just use RiskModels as the models for the view.

File: srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java [hunk 39]
```diff
@@ -0,0 +1,94 @@
...
+        	RiskModel rm = fAdminService.getRiskModelForId( riskModelId ); 
+
+        	if( rm == null ) {
+        		throw new InvalidIdentifierException( "Unable to find RiskModel with ID "+riskModelId );        		
+        	}
+        	EditRiskModel editModel = EditRiskModel.fromRiskModel( rm );
+	
+	        return editModel;    		
+    	}
+    	catch ( NumberFormatException nfe ) {
```
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelController.java]: Can this exception actually be thrown now? I don't think so.

File: srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidator.java (hunk 27 above)
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidator.java]: I still don't understand this comment.

File: srcalc/src/main/java/gov/va/med/srcalc/domain/model/DiscreteTerm.java (hunk 34 above)
Comment (gregoryhull) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/model/DiscreteTerm.java]: The BooleanTerm had a comment to this effect but DiscreteTerm didn't. Its not really a boolean so I thought this was a cut-n-paste error.  I think its clearer w/o the ? but obviously it doesn't matter much either way.

File: srcalc/src/main/java/gov/va/med/srcalc/domain/model/RiskModel.java (hunk 35 above)
Comment (gregoryhull) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/model/RiskModel.java]: Ok. . This was part of my trial-n-error attempts to fix the bug in the getAllRiskModels() query.

File: srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java (hunk 36 above)
Comment (gregoryhull) [inline srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java]: Shouldn't happen since there is currently no way for the user create or add new Ids.
Its just a sanity check / warning.

File: srcalc/src/main/java/gov/va/med/srcalc/service/ModelInspectionService.java (hunk 37 above)
Comment (gregoryhull) [inline srcalc/src/main/java/gov/va/med/srcalc/service/ModelInspectionService.java]: Moved to the AdminService.
I must have missed your earlier comment.

File: srcalc/src/test/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelControllerIT.java [hunk 40]
```diff
@@ -0,0 +1,96 @@
...
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.context.WebApplicationContext;
+
+/**
+ * Integration Test for {@link EditVariableController}. Only tests some basic
```
Comment (dgtombs) [inline srcalc/src/test/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelControllerIT.java]: Incorrect Javadoc. "EditVariableController"

File: srcalc/src/test/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelControllerIT.java [hunk 41]
```diff
@@ -0,0 +1,96 @@
...
+    public void testEditModel() throws Exception
+    {
+        fMockMvc.perform(get("/admin/models/2")).
+            andExpect(status().isOk()).
+            andExpect(model().attribute("riskModel", isA(EditRiskModel.class)));
+        
+        fMockMvc.perform(
+                post("/admin/models/2").
+                param("modelName", "Bogus Model Name")).
+            andExpect(redirectedUrl(AdminHomeController.BASE_URL));
```
Comment (dgtombs) [inline srcalc/src/test/java/gov/va/med/srcalc/web/controller/admin/EditRiskModelControllerIT.java]: Overall these tests are good, but you should actually verify that the DB was updated here. See `EditVariableControllerIT.testEditMultiSelectVariable()` for an example.

File: srcalc/src/test/java/gov/va/med/srcalc/web/view/admin/EditRiskModelTest.java [hunk 42]
```diff
@@ -0,0 +1,63 @@
...
+        
+    	editRiskModel.setModelName( "NewModelName" );
+    	
+        assertEquals( editRiskModel.getModelName(), "NewModelName" );
+        assertEquals( editRiskModel.getMaxDisplayNameLength(), RiskModel.DISPLAY_NAME_MAX );
+        
+        editRiskModel.applyChanges();
+        
+        RiskModel rm = editRiskModel.getRiskModel();
+        assertEquals("NewModelName", rm.getDisplayName() );
```
Comment (dgtombs) [inline srcalc/src/test/java/gov/va/med/srcalc/web/view/admin/EditRiskModelTest.java]: Good.

File: srcalc/src/test/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidatorTest.java [hunk 43]
```diff
@@ -0,0 +1,51 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import gov.va.med.srcalc.util.ValidationCodes;
+
+import org.junit.Test;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.BindingResult;
+
+public class EditRiskModelValidatorTest {
```
Comment (dgtombs) [inline srcalc/src/test/java/gov/va/med/srcalc/web/view/admin/EditRiskModelValidatorTest.java]: The formatting is inconsistent in this class. Can you have Eclipse reformat the file?

File: srcalc/src/test/java/gov/va/med/srcalc/web/view/admin/EditRiskModelTest.java [hunk 44]
```diff
@@ -0,0 +1,63 @@
...
+import gov.va.med.srcalc.service.MockModelService;
+
+import org.junit.Test;
+
+public class EditRiskModelTest {
+    private final MockModelService fModelService = new MockModelService();
+    
+    public static EditRiskModel createEditRiskModel( String name) 
+    {
+        final ProcedureVariable procedureVar = SampleModels.procedureVariable();
```
Comment (dgtombs) [inline srcalc/src/test/java/gov/va/med/srcalc/web/view/admin/EditRiskModelTest.java]: Is there any reason you didn't just use `SampleModels.thoracicRiskModel()`?

File: srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java (hunk 36 above)
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java]: OK.

File: srcalc/src/main/java/gov/va/med/srcalc/domain/model/DiscreteTerm.java (hunk 34 above)
Comment (dgtombs) [inline srcalc/src/main/java/gov/va/med/srcalc/domain/model/DiscreteTerm.java]: Even though it's not a "Boolean Term", it is still a boolean condition (A == B). This is as opposed to NumericalTerm, which has no boolean aspect to it at all.

It could go both ways, but we shouldn't change code that is already on master without a good reason. Please revert the change.
